bool x_movingForward
bool x_movingBackwards
bool x_movingLeft
bool x_movingRight
uint8 i_currentSpeed
